lib/ansible/modules/identity/cyberark/cyberark_authentication.py

```diff
@@ -0,0 +1,324 @@
+
+            if new_password is not None:
+                payload_dict["newPassword"] = new_password
+
+            payload = json.dumps(payload_dict)
+
+    else: # Logoff Action
+
+        # Get values from cyberark_session already established
+        api_base_url = cyberark_session["api_base_url"]
+        validate_certs = cyberark_session["validate_certs"]
+        use_shared_logon_authentication = cyberark_session[
+            "use_shared_logon_authentication"]
+        headers["Authorization"] = cyberark_session["token"]
+
+        # Different end_points based on the use of shared logon authentication
+        if use_shared_logon_authentication:
+            end_point = "/PasswordVault/WebServices/auth/Shared/RestfulAuthenticationService.svc/Logoff"
+        else:
+            end_point = "/PasswordVault/WebServices/auth/Cyberark/CyberArkAuthenticationService.svc/Logoff"
+
+    try:
+
+        response = open_url(
+            api_base_url + end_point,
+            method="POST",
+            headers=headers,
+            data=payload,
+            validate_certs=validate_certs)
+
+        result = None
+        changed = False
+
+        if response.getcode() == 200: # Success
+
+            if state == "present": # Logon Action
+
+                # Result token from REST Api uses a different key based
+                # the use of shared logon authentication
+                if use_shared_logon_authentication:
+                    token = json.loads(response.read())["LogonResult"]
+                else:
+                    token = json.loads(response.read())["CyberArkLogonResult"]
+
+                # Preparing result of the module
+                result = {
+                    "cyberark_session": {
+                        "token": token,
+                        "api_base_url": api_base_url,
+                        "validate_certs": validate_certs,
+                        "use_shared_logon_authentication":
+                        use_shared_logon_authentication},
+                }
+
+                if new_password is not None:
+                    # Only marks change if new_password was received resulting
+                    # in a password change
+                    changed = True
+
+            else: # Logoff Action clears cyberark_session
+
+                result = {
+                    "cyberark_session": {}
+                }
+
+            return (changed, result, response.getcode())
+
+        else:
+            module.fail_json(
+                msg="error in end_point=>" +
+                end_point +
+                json.dumps(headers))
+
+    except httplib.HTTPException:
+
+        t, e = sys.exc_info()[:2]
+        module.fail_json(
+            msg="end_point=" +
+            api_base_url +
+            end_point +
+            " headers=[" +
+            json.dumps(headers) +
+            "] payload=[" +
+            payload +
+            "] ===>" +
+            str(e),
```

Comment:
split headers and/or payload to different items in the fail_json? 

``` python
    fail_json(msg='asfad', headers=headers, payload=payload)
```

Suppose that depends if it would ever be useful on its own instead of part of the error msg.